### PR TITLE
[PW-3708] Update the quoteid for refuse payments

### PR DESCRIPTION
**Pull request by AlexandrosMor**
**Description**

Update the quoteid for refuse payments to avoid duplicated Quote IDs in sales_order_table

**Tested scenarios**

**Fixed issue**:  #915

### CODE REVIEW

**Review by msilvagarcia — DISMISSED** [review]

File: Controller/Process/Result.php [hunk 1]
```diff
@@ -117,6 +133,23 @@ public function execute()
                 $this->restoreCart($response);
                 $failReturnPath = $this->_adyenHelper->getAdyenAbstractConfigData('return_path');
                 $this->_redirect($failReturnPath);
+
+                $currectQuote =  $session->getQuote();
```
Comment (acampos1916) [inline Controller/Process/Result.php:137]:
```suggestion
                $currentQuote =  $session->getQuote();
```

File: Controller/Process/Result.php [hunk 2]
```diff
@@ -117,6 +133,23 @@ public function execute()
...
+                if (!empty($currectQuote)) {
```
Comment (acampos1916) [inline Controller/Process/Result.php:138]:
```suggestion
                if (!empty($currentQuote)) {
```

File: Controller/Process/Result.php [hunk 3]
```diff
@@ -117,6 +133,23 @@ public function execute()
...
+                    /**
+                     * @var \Magento\Quote\Model\Quote $newQuote
+                     */
+                    $newQuote = $this->quoteFactory->create();
+                    $currentQuoteId = $currectQuote->getId();
```
Comment (acampos1916) [inline Controller/Process/Result.php:143]:
```suggestion
                    $currentQuoteId = $currentQuote->getId();
```

File: Controller/Process/Result.php (hunk 1 above)
Comment (AlexandrosMor) [reply Controller/Process/Result.php:137]: ops :)

**Review by peterojo — APPROVED** [review]

File: Controller/Process/Result.php [hunk 4]
```diff
@@ -117,6 +133,23 @@ public function execute()
...
+                    /**
+                     * @var \Magento\Quote\Model\Quote $newQuote
+                     */
+                    $newQuote = $this->quoteFactory->create();
+                    $currentQuoteId = $currentQuote->getId();
+
+                    $newQuote->merge($currectQuote);
+                    $newQuote->setId($currentQuoteId + 1);
+                    //Close the oldQuote and set the new
+                    $currectQuote->setIsActive(false);
```
Comment (msilvagarcia) [inline Controller/Process/Result.php:148]:
```suggestion
                    $currentQuote->setIsActive(false);
```

File: Controller/Process/Result.php [hunk 5]
```diff
@@ -117,6 +133,23 @@ public function execute()
...
+
+                $currentQuote =  $session->getQuote();
+                if (!empty($currentQuote)) {
+                    /**
+                     * @var \Magento\Quote\Model\Quote $newQuote
+                     */
+                    $newQuote = $this->quoteFactory->create();
+                    $currentQuoteId = $currentQuote->getId();
+
+                    $newQuote->merge($currectQuote);
```
Comment (msilvagarcia) [inline Controller/Process/Result.php:145]:
```suggestion
                    $newQuote->merge($currentQuote);
```

File: Controller/Process/Result.php [hunk 6]
```diff
@@ -117,6 +133,23 @@ public function execute()
...
+                    $newQuote->setId($currentQuoteId + 1);
+                    //Close the oldQuote and set the new
+                    $currectQuote->setIsActive(false);
+                    $newQuote->setIsActive(true);
+                    $this->cartRepository->save($newQuote);
+                    $this->cartRepository->save($currectQuote);
```
Comment (msilvagarcia) [inline Controller/Process/Result.php:151]:
```suggestion
                    $this->cartRepository->save($currentQuote);
```

**Review by msilvagarcia — CHANGES_REQUESTED** [review]

**Review by acampos1916 — APPROVED** [review]

**Review by peterojo — APPROVED** [review]

**Review by msilvagarcia — APPROVED** [review]

**Pull request merged**